File: Wordplay/python/12.py
# Create and print an array containing all of the words that end in "GHTLY"
def main():
    in_file = open ("../sowpods.txt", "r")
    ghtly_arr = []
    for words in in_file:
        word_len = len(words) - 1 # -1 for escape character
        # Check if word is elibible to meet conditions
        if("GHTLY" not in words):
            continue
        # Check if last 5 chars of word contain 'GHTLY'
        elif(words[word_len - 5: word_len] != "GHTLY"):
            continue
        # If all checks passed, append to list
        ghtly_arr.append(words)
    print(ghtly_arr, end="")

if __name__ == '__main__':
    main()

File: Wordplay/js/12.js
// Create and print an array containing all of the words that end in "GHTLY"

let main = () => {

    const fs = require("fs");
    let buffer_arry = fs.readFileSync("Wordplay/sowpods.txt", "utf-8");
    let in_file = buffer_arry.split("\n");
    let ghtly_arry = [];

    // Function checks if word ends in 'GHTLY'
    let check_ghtly = (word) => {
        
        if(word.slice(-5) == "GHTLY") {
            ghtly_arry.push(word);
        };
    };

    // Function checks if words is long enough
    let check_len = (word) => {
        if(word.length < 5) {
            return false;
        };
        return true;
    };

    // Main function handling
    for(let i = 0; i < in_file.length; i++) {
        if(check_len(in_file[i])) {
            check_ghtly(in_file[i]);
        };
    };

    console.log(ghtly_arry);
};

main();